WebContent/WEB-INF/jsp/dataSourceEdit/editModbusIp.jsp

```diff
@@ -55,13 +55,95 @@
       DataSourceEditDwr.testModbusIpData($get("timeout"), $get("retries"), $get("transportType"), $get("host"), 
               $get("port"), $get("encapsulated"), slaveId, range, offset, length, dataTestCB);
   }
+
+  function validateModbusProperties(temp){
+
+    let messages = [];
+
+    validateValue("updatePeriods", "<fmt:message key='badIntegerFormat'/>", isPositiveInt, temp.updatePeriods, messages);
+    validateValue("updatePeriodType", "<fmt:message key='badIntegerFormat'/>", isPositiveInt, temp.updatePeriodType, messages);
+    validateValue("timeout", "<fmt:message key='badIntegerFormat'/>", isPositiveInt, temp.timeout, messages);
+    validateValue("retries", "<fmt:message key='badIntegerFormat'/>", isPositiveInt, temp.retries, messages);
+    validateValue("maxReadBitCount", "<fmt:message key='badIntegerFormat'/>", isPositiveInt, temp.maxReadBitCount, messages);
+    validateValue("maxReadRegisterCount", "<fmt:message key='badIntegerFormat'/>", isPositiveInt, temp.maxReadRegisterCount, messages);
+    validateValue("maxWriteRegisterCount", "<fmt:message key='badIntegerFormat'/>", isPositiveInt, temp.maxWriteRegisterCount, messages);
+    validateValue("port", "<fmt:message key='badIntegerFormat'/>", isPositiveInt, temp.port, messages);
+
+    return messages;
+  }
+
+  function createTempModbusData(){
+
+    let dataSourceName = $get("dataSourceName");
```

Comment:
There is no point in creating additional variables here, you get and immediately set them in the temp structure, you don't parse here, etc.